Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANS/SANSCreateAdjustmentWorkspaces.py

```diff
@@ -0,0 +1,228 @@
+                                                     direction=Direction.Output),
+                             doc='The workspace for wavelength-based adjustments.')
+
+        self.declareProperty(MatrixWorkspaceProperty('OutputWorkspacePixelAdjustment', '',
+                                                     direction=Direction.Output),
+                             doc='The workspace for wavelength-based adjustments.')
+
+        self.declareProperty(MatrixWorkspaceProperty('OutputWorkspaceWavelengthAndPixelAdjustment', '',
+                                                     direction=Direction.Output),
+                             doc='The workspace for, both, wavelength- and pixel-based adjustments.')
+
+    def PyExec(self):
+        # Read the state
+        state_property_manager = self.getProperty("SANSState").value
+        state = create_deserialized_sans_state_from_property_manager(state_property_manager)
+
+        # --------------------------------------
+        # Get the monitor normalization workspace
+        # --------------------------------------
+        monitor_normalization_workspace = self._get_monitor_normalization_workspace(state)
+
+        # --------------------------------------
+        # Get the calculated transmission
+        # --------------------------------------
+        calculated_transmission_workspace, unfitted_transmission_workspace =\
+            self._get_calculated_transmission_workspace(state)
+
+        # --------------------------------------
+        # Get the wide angle correction workspace
+        # --------------------------------------
+        wave_length_and_pixel_adjustment_workspace = self._get_wide_angle_correction_workspace(state,
+                                                                   calculated_transmission_workspace)  # noqa
+
+        # --------------------------------------------
+        # Get the full wavelength and pixel adjustment
+        # --------------------------------------------
+        wave_length_adjustment_workspace, \
+        pixel_length_adjustment_workspace = self._get_wavelength_and_pixel_adjustment_workspaces(state,
+                                                                            monitor_normalization_workspace,  # noqa
+                                                                            calculated_transmission_workspace)  # noqa
+
+        if wave_length_adjustment_workspace:
+            self.setProperty("OutputWorkspaceWavelengthAdjustment", wave_length_adjustment_workspace)
+        if pixel_length_adjustment_workspace:
+            self.setProperty("OutputWorkspacePixelAdjustment", pixel_length_adjustment_workspace)
+        if wave_length_and_pixel_adjustment_workspace:
+            self.setProperty("OutputWorkspaceWavelengthAndPixelAdjustment", wave_length_and_pixel_adjustment_workspace)
+
+        # TODO set diagnostic output workspaces
```

Comment:
Yes, there are several TODOs in there which refer to "nice-to-have" features which we can add once we have freed up resources. I will expand the TODOs